### PR TITLE
Inline setup_external_compile into setupext.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Last chunk of breaking #11570 into smaller pieces.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: setupext.py [hunk 1]
```diff
@@ -1124,26 +1124,30 @@ def do_custom_build(self):
...
-            cmdfile = os.path.join("build", "build_freetype.cmd")
-            with open(cmdfile, 'w') as cmd:
-                cmd.write(prepare_build_cmd(FREETYPE_BUILD_CMD, vc20xx=vc, WinXX=WinXX,
-                                            config='Release'))
-
-            shutil.rmtree(str(Path(src_path, "objs")), ignore_errors=True)
-            subprocess.check_call([os.path.abspath(cmdfile)],
+            if platform.architecture()[0] == '64bit':
+                WinXX = xXX = 'x64'
+            else:
```
Comment (timhoffm) [inline setupext.py]: Maybe just use `.. = .. if platform.architecture()[0] else ..` twice.

File: setupext.py [hunk 2]
```diff
@@ -1124,26 +1124,30 @@ def do_custom_build(self):
...
+                WinXX = 'Win32'
+                xXX = 'x86'
+            vcvarsall = msvc.find_vcvarsall(10.0)
+            if vcvarsall == None:
```
Comment (timhoffm) [inline setupext.py]: `is None`

File: setupext.py [hunk 3]
```diff
@@ -1124,26 +1124,30 @@ def do_custom_build(self):
...
+                raise RuntimeError('Microsoft VS 2010 required')
+            cmdfile = Path("build/build_freetype.cmd")
+            cmdfile.write_text(
+                FREETYPE_BUILD_CMD.format(vc20xx=vc, WinXX=WinXX, xXX=xXX))
```
Comment (timhoffm) [inline setupext.py]: `vcvarsall` missing in format

File: setupext.py (hunk 1 above)
Comment (anntzer) [reply setupext.py]: done

File: setupext.py (hunk 3 above)
Comment (anntzer) [reply setupext.py]: done

File: setupext.py (hunk 2 above)
Comment (anntzer) [reply setupext.py]: done

**Pull request merged**